### ISSUE TITLE
Does ecos really have the same signature as the others?

### ISSUE THREAD

**Issue by microprediction** [stephane-caron/qpsolvers#165]
When I set up this problem all the solvers swallow it but ecos complains:

        "'all the input arrays must have same number of dimensions, but the '
         'array at index 0 has 2 dimension(s) and the array at index 1 has 1 '
         'dimension(s)',"
       
The problem:
   
            def get_fin_problem(n_dim=5) -> Problem:
                print("    min. 1/2 x^T P x + q^T x")
                print("    s.t. G * x <= h")
                print("         A * x == b")
                P = random_factor_cov(n_dim=n_dim)   # <---  doesn't matter returns n_dim, n_dim
                q = 0.01*np.ones(shape=(n_dim,1))
                G = np.eye(n_dim)
                h = np.ones(shape=(n_dim,))
                A = np.ones(shape=(n_dim,))
                b = np.ones(shape=(1,))
                return Problem(P, q, G, h, A, b)

**Comment by stephane-caron** [comment]
I can reproduce the issue with:

```python
import numpy as np

from qpsolvers import Problem, solve_problem

n_dim = 3
P = np.eye(n_dim)
q = 0.01 * np.ones(shape=(n_dim, 1))
G = np.eye(n_dim)
h = np.ones(shape=(n_dim,))
A = np.ones(shape=(n_dim,))
b = np.ones(shape=(1,))
problem = Problem(P, q, G, h, A, b)

solution = solve_problem(problem, solver="ecos")
```

**Comment by stephane-caron** [comment]
The issue comes from an internal function expecting `q` to be flat. I'll push a fix to make the function flatten vectors if applicable. In the meantime you can flatten your input to use the solver:

```python
q = q.flatten()
```

**Comment by stephane-caron** [comment]
This issue should be fixed in the latest [v2.7.4](https://github.com/stephane-caron/qpsolvers/releases/tag/v2.7.4) (just released). Hoping it helps!

**Comment by microprediction** [comment]
You are incredibly responsive. Many thanks.
